src/test/java/org/mockito/internal/matchers/CustomCapturingMatcherTest.java

```diff
@@ -0,0 +1,46 @@
+package org.mockito.internal.matchers;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameters;
+
+@RunWith(Parameterized.class)
+public class CustomCapturingMatcherTest {
+
+	private boolean actual, expected;
+
+	public CustomCapturingMatcherTest(final boolean actual, final boolean expected) {
+		this.actual = actual;
+		this.expected = expected;
+	}
+
+	@Test
+	public void verify_scenario() {
+		assertThat(expected, is(equalTo(actual)));
+	}
+
+	@Parameters
+	public static List<Object[]> scenarios() {
+		final List<Object[]> scenarios = new ArrayList<Object[]>();
+		scenarios.add(getParams(CustomCapturingMatcher.is(new InstanceOf(String.class)).matches("string_type"), true));
```

Comment:
`CustomCapturingMatcher.is(ArgumentMatcher)` is fine! I mean you should consider to call eg.:

scenarios.add(getParams(CustomCapturingMatcher.is(isA(String.class)).matches(1), false));